src/main/java/org/openrewrite/java/migrate/io/RemoveFinalizeFromFileStream.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.openrewrite.java.migrate.io;
+
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import org.openrewrite.ExecutionContext;
+import org.openrewrite.Preconditions;
+import org.openrewrite.Recipe;
+import org.openrewrite.TreeVisitor;
+import org.openrewrite.java.JavaVisitor;
+import org.openrewrite.java.MethodMatcher;
+import org.openrewrite.java.search.UsesJavaVersion;
+import org.openrewrite.java.search.UsesType;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+import org.openrewrite.java.tree.TypeUtils;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@Value
+@EqualsAndHashCode(callSuper = true)
+public class RemoveFinalizeFromFileStream extends Recipe {
+
+    private static final String JAVA_IO_FILEINPUTSTREAM = "java.io.FileInputStream";
+    private static final String JAVA_IO_FILEOUTPUTSTREAM = "java.io.FileOutputStream";
+
+    private static final MethodMatcher METHOD_MATCHER = new MethodMatcher("java.lang.Object finalize()");
+
+    @Override
+    public String getDisplayName() {
+        return "Replace invocations of deprecated invocations from FileInputStream, FileOutputStream";
+    }
+
+    @Override
+    public String getDescription() {
+        return "Remove invocations of finalize() deprecated invocations from FileInputStream, FileOutputStream.";
+    }
+
+    @Override
+    public Set<String> getTags() {
+        return Collections.singleton("JDK-8212050");
+    }
+
+    @Override
+    public TreeVisitor<?, ExecutionContext> getVisitor() {
+
+        return Preconditions.check(
+                Preconditions.and(
+                        new UsesJavaVersion<>(9),
```

Comment:
Since these methods were removed in Java 12, we don't need to check there anymore:
```suggestion
                        new UsesJavaVersion<>(9, 11),
```

Reply:
fixed